### PR TITLE
Add hazelcast-it module with integration tests to OS

### PR DESCRIPTION
**Pull request by frant-hartm**
We already have hazelcast-it module with integration tests in EE.
See https://github.com/hazelcast/hazelcast-enterprise/pull/4402.

This EE module will be renamed to avoid the name clash - (https://github.com/hazelcast/hazelcast-enterprise/pull/4686).

This module should contain integration tests, which expect a clean
environment (e.g. classpath without many dependencies) etc..

We add this module to OS as well to make it easier to write tests for OS
PRs - we don't have to submit 2 PRs, one to OS, one to EE.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-it/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>add-third-party</id>
+                        <phase>none</phase>
```
Comment (ldziedziul) [inline hazelcast-it/pom.xml:55]: Why do we need to disable it?

File: hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java [hunk 2]
```diff
@@ -1611,7 +1611,7 @@ public static void assumeThatNotZingJDK6() {
     }
 
     public static void assumeThatNoWindowsOS() {
-        assumeFalse(lowerCaseInternal(System.getProperty("os.name")).contains("windows"));
+        assumeFalse(OsHelper.isWindows());
```
Comment (ldziedziul) [inline hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java:1614]: Maybe: 
```suggestion
        assumeFalse("Skipping on Windows", OsHelper.isWindows());
```

**Review by ldziedziul — APPROVED** [review]
Looks good, just 2 comments

File: hazelcast-it/pom.xml (hunk 1 above)
Comment (frant-hartm) [reply hazelcast-it/pom.xml:55]: I actually don't remember why I had it in the original PR.

Tried without it and it seems to work fine.

File: hazelcast-it/distribution-it/pom.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    </parent>
+
+    <artifactId>distribution-it</artifactId>
+    <version>5.1-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.hazelcast</groupId>
+            <artifactId>hazelcast</artifactId>
+            <version>5.1-SNAPSHOT</version>
```
Comment (olukas) [inline hazelcast-it/distribution-it/pom.xml:19]: we should not hard-coded version, please use `${project.version}`. The same for some other dependencies in this pom.xml and in another pom.xml in this PR.

File: hazelcast-it/pom.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    <description>Hazelcast Integration Tests</description>
+
+    <modules>
+        <module>distribution-it</module>
+        <!-- See other modules in profiles -->
+    </modules>
+
+    <properties>
+        <!-- needed for CheckStyle -->
+        <main.basedir>${project.basedir}/../..</main.basedir>
```
Comment (olukas) [inline hazelcast-it/pom.xml:40]: Is this correct? It doesn't seem as correct path from this level. If it is expected to be used from submodule I think it should be rather defined there. WDYT?

**Review by olukas — CHANGES_REQUESTED** [review]
LGTM but hard-coded HZ version has to be removed.

File: hazelcast-it/pom.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    </properties>
+
+    <dependencies>
+
+    </dependencies>
```
Comment (mdumandag) [inline hazelcast-it/pom.xml:45]: Empty tag

File: hazelcast-it/jdk17-tests/pom.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <plugins>
+        </plugins>
+    </build>
```
Comment (mdumandag) [inline hazelcast-it/jdk17-tests/pom.xml:39]: Empty tag

File: hazelcast-it/distribution-it/pom.xml (hunk 3 above)
Comment (frant-hartm) [reply hazelcast-it/distribution-it/pom.xml]: Fixed.
Only hardcoded versions left are in the parent sections of the poms.

File: hazelcast-it/pom.xml (hunk 4 above)
Comment (frant-hartm) [reply hazelcast-it/pom.xml:40]: Yes, this is intentional. Similar to pom.xml in extensions
https://github.com/hazelcast/hazelcast/blob/master/extensions/pom.xml#L83
or the ee version
https://github.com/hazelcast/hazelcast-enterprise/blob/master/hazelcast-it/pom.xml#L47

It works for the direct submodules (most common case), if you have more complicated structure than this needs to be changed at the particular level.

File: hazelcast-it/pom.xml (hunk 5 above)
Comment (frant-hartm) [reply hazelcast-it/pom.xml]: Added a comment (same as in EE https://github.com/hazelcast/hazelcast-enterprise/blob/master/hazelcast-it/pom.xml#L52-L57).

File: hazelcast-it/jdk17-tests/pom.xml (hunk 6 above)
Comment (frant-hartm) [reply hazelcast-it/jdk17-tests/pom.xml]: Removed.

**Review by olukas — APPROVED** [review]

**Review by mdumandag — APPROVED** [review]

**Pull request merged**